AirCasting/APICommunicator/StreamDownloader.swift

```diff
@@ -0,0 +1,74 @@
+// Created by Lunar on 29/03/2022.
+//
+
+import Foundation
+import Resolver
+
+struct StreamWithMeasurementsDownstream: Decodable {
+    let title: String
+    let username: String
+    let measurements: [StreamWithMeasurementsDownstream.Measurements]
+    let id: Int
+    let lastMeasurementValue: Double
+    let sensorName: String
+    let sensorUnit: String
+    
+    struct Measurements: Decodable {
+        let value: Double
+        let time: Int
+        let longitude: Double
+        let latitude: Double
+    }
+}
+
+protocol StreamDownloader {
+    func downloadStreamWithMeasurements(id: String, completion: @escaping (Result<StreamWithMeasurementsDownstream, Error>) -> Void)
+}
+
+class DefaultStreamDownloader: StreamDownloader {
+    @Injected private var urlProvider: URLProvider
+    @Injected private var apiClient: APIClient
+    @Injected private var responseValidator: HTTPResponseValidator
+    @Injected private var authorisationService: RequestAuthorisationService
+    
+    private let decoder: JSONDecoder = {
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+        return decoder
+    }()
+    
+    func downloadStreamWithMeasurements(id: String, completion: @escaping (Result<StreamWithMeasurementsDownstream, Error>) -> Void) {
+        let urlComponentPart = urlProvider.baseAppURL.appendingPathComponent("api/fixed/streams/\(id).json")
+        var urlComponents = URLComponents(string: urlComponentPart.absoluteString)!
+        urlComponents.queryItems = [
+            URLQueryItem(name: "measurements_limit", value: "1140")
```

Comment:
I think the limit value should be passed in. Either via func parameter or an init value for the whole service.